tests/ui/match/mut-ref-mut-2021.stderr

```diff
@@ -0,0 +1,70 @@
+error[E0384]: cannot assign twice to immutable variable `a`
+  --> $DIR/mut-ref-mut-2021.rs:7:5
+   |
+LL |     let Foo(a) = Foo(0);
+   |             -
+   |             |
+   |             first assignment to `a`
+   |             help: consider making this binding mutable: `mut a`
```

Comment:
it would be nice to extend this help message to suggest `mut ref` as well when the feature is active. I'lll let you decide whether to include in this PR or do it separately

Reply:
I'll do it separately, I'd rather the `BindingAnnotation`/`ByRef` changes land first so later PRs can make use of them